clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -133,7 +134,7 @@ public void onBecomeLeaderFromStandby(Message message, NotificationContext conte
      * where a slice doesn't have a replicating leader.  While this state transition executes, there should be no
      * other leader in the slice.
      */
-    updateLagMonitor(message.getResourceName(), heartbeatMonitoringService::addLeaderLagMonitor);
+    updateLagMonitor(message.getResourceName(), heartbeatMonitoringService::addLeaderLagMonitor, false);
```

Comment:
A race between inflight ST and version retirement is possible. If version is deleted first, then we'll consider `isNullVersionValid` as false which may not be true?

Reply:
In this scenario I think we should just log the error and do nothing instead of trying to add leader lag monitor for a store version that will soon be deleted (LEADER/STANDBY->OFFLINE). Such race should be rare too, we can make incremental changes. If this really becomes common or a problem we should implement a better mechanism for this.

Reply:
That sounds reasonable. Thank!